### PR TITLE
Implement ReleaseVm directly

### PR DESCRIPTION
**Pull request by Crazykev**
/cc @laijs 

Signed-off-by: Crazykev <crazykev@zju.edu.cn>

### CODE REVIEW

File: hypervisor/vm.go [hunk 1]
```diff
@@ -130,24 +130,20 @@ func (vm *Vm) WaitResponse(match matchResponse, timeout int) chan error {
...
-			return nil, true
-		}
-		return nil, false
-	}, -1)
-
-	releasePodEvent := &ReleaseVMCommand{}
+	orig := ctx.current
+	ctx.lock.Lock()
+	ctx.current = StateNone
+	ctx.handler = nil
```
Comment (laijs) [inline hypervisor/vm.go]: ctx.loop() will access to ctx.handler without lock, so this code my cause ctx.loop() pannic.

Comment (Crazykev) [inline hypervisor/vm.go]: remove lock here won't help this, should I add a lock in ctx.loop() ?

**Comment by allencloud** [comment]
I am afraid conflict happens. A rebase needed though. @Crazykev

File: hypervisor/hypervisor.go [hunk 2]
```diff
@@ -20,8 +20,14 @@ func (ctx *VmContext) loop() {
 			glog.V(1).Info("got nil event.")
 			continue
 		}
+		ctx.lock.Lock()
+		if ctx.handler == nil {
+			ctx.lock.Unlock()
```
Comment (allencloud) [inline hypervisor/hypervisor.go]: why not use a defer to simplify the code?

Comment (Crazykev) [reply hypervisor/hypervisor.go]: @allencloud I only want to lock `ctx.handler(ctx, ev)` here, if we use lock and defer, the whole event loop will be in the lock, especially waiting an event from `ev, ok := <-ctx.Hub`.

BTW, I'm not sure this patch is still on the correct direction. /cc @laijs

**Pull request closed without merging**